### PR TITLE
fully move jenkins plugin PR jobs from jenkins based aos-cd-jobs to p…

### PR DESCRIPTION
**Pull request by gabemontero**
…row based openshift/release

/assign @bparees 
/assign @stevekuznetsov 

@openshift/sig-developer-experience fyi

### CODE REVIEW

File: ci-operator/config/openshift/jenkins-client-plugin/openshift-jenkins-client-master.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+tag_specification:
+  name: '4.0'
+  namespace: ocp
+promotion:
+  name: '4.0'
+  namespace: ocp
```
Comment (bparees) [inline ci-operator/config/openshift/jenkins-client-plugin/openshift-jenkins-client-master.yaml]: i don't think you want promotion on this, right?

Comment (bparees) [reply ci-operator/config/openshift/jenkins-client-plugin/openshift-jenkins-client-master.yaml]: (you're building a dummy jenkins image w/ a patched plugin for purposes of running the e2e, you don't want that to be promoted for general consumption).

Comment (gabemontero) [reply ci-operator/config/openshift/jenkins-client-plugin/openshift-jenkins-client-master.yaml]: yeah I read the doc on that in ci-operator but still wan't entirely sure of the implication .. `configures how images that are built from the repository under test are promoted for use in tests for other repositories`  ... at the time it was unclear to me how far things went to enable "other repositories"

I'll remove

**Comment by gabemontero** [comment]
Tests like 
```
./ci-operator --config ../release/ci-operator/config/openshift/jenkins-client-plugin/openshift-jenkins-client-master.yaml  --artifact-dir /home/gmontero/tmpsave --namespace gabemontero-plugin-test --git-ref openshift/jenkins-client-plugin@master
```

looked good from a consumption of the new plugin dockefiles

**Comment by gabemontero** [comment]
Also, reading the ci-operator doc, I don't think I need `canonical_go_repository` either but was hesitant to diverge and it seemed benign enough

**Comment by gabemontero** [comment]
still sorting out the mechanics of the plugin Dockerfiles to build the jenkins image to be tested, but have pushed changes to attempt to address the various check failures

**Comment by gabemontero** [comment]
ok I got my local runs of ci-operator using the new config files against api.ci working now ... (using `base` as the original jenkins base_image no bueno)

**Comment by gabemontero** [comment]
So some of the rehearsal runs are still in progress, but some of them are hitting the recent auth/509cert bugs when the e2e ext ginkgo test tries to set up its client

https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_release/3177/rehearse-3177-pull-ci-openshift-jenkins-sync-plugin-master-e2e-aws-jenkins/3 for example

I saw these as well with recent e2e-image-registry runs, though you can occasionally avoid them 

But the image build and launching of the jenkins e2e's looks good (and is  the crux of this PR

**Comment by gabemontero** [comment]
these rehearse jobs are  very cool btw dev productivity team :-)

**Comment by gabemontero** [comment]
yep some jobs did in fact get pass unrelated flakes ... see this client plugin e2e run:  https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_release/3177/rehearse-3177-pull-ci-openshift-jenkins-openshift-login-plugin-master-e2e-aws-jenkins/3

**Comment by gabemontero** [comment]
the known unauthorized/x509/certs flakes being tracked by buildcop

/retest

**Comment by gabemontero** [comment]
bump @bparees @stevekuznetsov 

fyi ... I've seen clean rehearsal runs for the all 3 plugins, just not at all the same time

**Pull request merged**